services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -325,7 +338,7 @@ public Integer getUsersCount(@QueryParam("search") String search,
             }
         }
 
-        List<UserModel> userModels = session.users().searchForUser(attributes, realm, firstResult, maxResults);
+        List<UserModel> userModels = searchImpl.get();
```

Comment:
Instead of all these changes to `UsersResource` isn't better to just change `` and call the expected method depending on the `SEARCH` filter? Something like this:

```
if (attributes.containsKey(UserModel.SEARCH)) {
                    return ((UserQueryProvider)provider).searchForUser(attributes.get(UserModel.SEARCH), realm, first, max);
                } else {
                    return ((UserQueryProvider)provider).searchForUser(attributes, realm, first, max);                    
                }
```

Reply:
@pedroigor +1, Updated PR per your suggestion.